src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -245,12 +260,14 @@ class MarathonSchedulerService @Inject() (
     timer = newTimer()
     oldTimer.cancel()
 
-    if (driver.isDefined) {
+    driver.map { driverInstance =>
+      mesosHeartbeatActor ! Heartbeat.MessageDeactivate(MesosHeartbeatMonitor.sessionOf(driverInstance))
       // Our leadership has been defeated. Thus, stop the driver.
-      // Note that abdication command will be ran upon driver shutdown which
-      // will then offer leadership again.
       stopDriver()
-    } else {
+    }
+    // Abdication will have already happened if the driver terminated abnormally.
+    // Otherwise we've either been terminated or have lost leadership for some other reason (network part?)
+    if (driver.isEmpty && isRunningLatch.getCount > 0) {
```

Comment:
the check for `driver.isEmpty` here is useless since `stopDriver` is guaranteed to set `driver` to `None`